Checkout: Show cart to the right of payments box

niranjan-uma-shankar opened the pull request:
#### Changes proposed in this Pull Request

* Following the positive results in the `showCheckoutCartRight` abtest run in #32832  we will make the cart on the right as the new control. For more context, also check p8Eqe3-Bo-p2.

Visiting the checkout page(either with a WordPress.com or Jetpack product), the cart will now always be shown to the right of the payments box/domain contact form. 

#### Testing instructions

#### Scenario 1
1. Open the checkout page by adding a new domain and plan to cart.
2. Verify that the checkout page looks like in the screenshot below:
     * Cart is to the right
     * The left sidebar is hidden.
     * The cart box and domain contact form box are of equal height and aligned.
     * The domain contact form and the cart are center aligned to the page

![calypso localhost_3000_checkout_xaf993401859 wordpress com](https://user-images.githubusercontent.com/1269602/58864646-bc2f9580-86d2-11e9-90cf-887f24c7548c.png)

#### Scenario 2

After completing steps 1-3 above, proceed to the Checkout page and verify it looks like in the screenshot below. Few things to look out for:
    * The checkout box and the cart box height are equal and aligned.
    * The left sidebar is hidden.
    * The header _Great Choice! How would you like to pay?_ is center aligned to the page.
    * The checkout box and cart box are center aligned to the page.

![calypso localhost_3000_checkout_xaf993401859 wordpress com (1)](https://user-images.githubusercontent.com/1269602/58864674-d0739280-86d2-11e9-9452-137b5c4ed611.png)

#### Scenario 3

1. On a site having a paid plan and _no domain_, purchase a domain and proceed to checkout.
2. The checkout page should like in the screenshot below. As before, verify the UI looks proper (cart to the right, boxes aligned and so on).

![calypso localhost_3000_checkout_xaf993401859 wordpress com (2)](https://user-images.githubusercontent.com/1269602/58864706-dc5f5480-86d2-11e9-9db1-7be63b6e42fc.png)

#### Scenario 4

Open Reader and My Sites pages. Verify that the sidebar looks fine. Adding this scenario since this PR modifies the <Sidebar> component which is being used in other Calypso pages.


#### Scenario 5

Verify that the placeholder components showed while the page is loading (the ones with the flashing grey horizontal bars in the gif below) show cart on the right with proper alignment.

![giphy (1)](https://user-images.githubusercontent.com/1269602/58864764-f731c900-86d2-11e9-8225-7db582e52a84.gif)

#### Scenario 6

Add a Jetpack plan to your cart and verify that Jetpack checkout shows cart to the right.

![calypso localhost_3000_checkout_niranjan ngrok io (1)](https://user-images.githubusercontent.com/1269602/58865450-3280c780-86d4-11e9-8eac-98de2a3a5cb0.png)

#### Scenario 7

Try the domains only flow starting at http://calypso.localhost:3000/start/domain and verify that you are able to complete the purchase. 

#### Scenario 8

**Verify on mobile devices:**

**iPhone X (Jetpack):**

![jp_mob copy](https://user-images.githubusercontent.com/1269602/58865910-0a459880-86d5-11e9-955e-0216f720ac13.png)

**Galaxy S5:**

![wpcom_right](https://user-images.githubusercontent.com/1269602/58866004-3bbe6400-86d5-11e9-8ffc-5a20130644cd.png)

**iPad Pro:**

![ipad_right](https://user-images.githubusercontent.com/1269602/58866025-47aa2600-86d5-11e9-8532-318d49052c6c.png)

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

**With Chat window open**

Desktop:

![calypso localhost_3000_checkout_niranjanumashankartest1 wordpress com (1)](https://user-images.githubusercontent.com/1269602/58866311-f484a300-86d5-11e9-8625-701296587362.png)

iPad:

![calypso localhost_3000_checkout_niranjanumashankartest1 wordpress com(iPad Pro) (1)](https://user-images.githubusercontent.com/1269602/58866323-fbabb100-86d5-11e9-9172-404b001b9cc7.png)

niranjan-uma-shankar committed:
First commit: Removing the showCartRight abtest and making the cart to the right as the control

niranjan-uma-shankar commented on client/my-sites/checkout/cart/secondary-cart.jsx:
```diff
@@ -78,16 +77,16 @@ class SecondaryCart extends Component {
 
 		return (
-			<Sidebar className={ cartClasses } hasSidebar={ this.props.hasSidebar }>
+			<ul className={ cartClasses }>
```
Just to explain the change here - I'm removing the `Sidebar` parent since after moving cart to the right, it no longer makes sense to call the cart a `Sidebar`.

victorespigares commented:
LGTM! 👍

victorespigares reviewed (approved):
Design-wise LGTM! 👍

dzver commented on client/my-sites/checkout/checkout/secure-payment-form.jsx:
```diff
@@ -542,13 +540,8 @@ export class SecurePaymentForm extends Component {
-		}
-		return <FormattedHeader headerText={ headerText } />;
+		const element = document.getElementsByClassName( 'formatted-header__title' )[ 0 ];
+		element.textContent = headerText;
```
Could you add a bit of info why we do it this way instead of the React way?

niranjan-uma-shankar replied:
You're right @dzver , this needs to change. I've implemented the React way in [3b4b80e](https://github.com/Automattic/wp-calypso/pull/33560/commits/0d61e898455e789f98ecebf359624a01fa259803).

dzver reviewed (approved):
LGTM

Left one note for a change that doesn't originate from this PR but from a previous one.

niranjan-uma-shankar committed:
Handling state changes for the header text

This commit fixes an issue of the header text being modified directly from the child using Node.textContent. Here, we introduce a new ChildContainer component that wraps around the Checkout component, and also handles the state transition of the header text.

niranjan-uma-shankar committed:
Restoring coupon code from merge

niranjan-uma-shankar committed:
Make the domain only flow to call the checkout() method and remove the sitelessCheckout() function. This is because both functions serve the same purpose so code repetition can be avoided.

niranjan-uma-shankar commented:
I'm not clear on the ESLint configuration error. I'll post an update when I have more details on what's causing it to break.

niranjan-uma-shankar commented:
Update: confirmed that the ESLint error is unrelated to my changes.

jsnajdr commented:
@niranjan-uma-shankar I have a fix for the ESLint in #33787.

dzver reviewed (commented):
LGTM

Looks much better after the refactoring, thanks!

niranjan-uma-shankar commented:
Thanks for the update @jsnajdr

delputnam commented:
LGTM @niranjan-uma-shankar. Thanks for the ping for review.
